src/commands/add-endpoint.js

```diff
@@ -1,32 +1,32 @@
 const {cli} = require('cli-ux')
+const { flags } = require('@oclif/command')
 const {saveEndpointOrError, errorOut} = require('../lib/misc.js')
 const FaunaCommand = require('../lib/fauna-command.js')
 const url = require('url')
 
 class AddEndpointCommand extends FaunaCommand {
   async run() {
     const endpoint = this.args.endpoint
+    var secret = this.flags.key
+    var alias = this.flags.alias
     const log = this.log
 
     const newEndpoint = url.parse(endpoint)
     if (!newEndpoint.hostname) {
       throw new Error('You must provide a valid endpoint.')
     }
 
-    return cli.prompt('Endpoint Key', {type: 'hide', timeout: 120000})
-    .then(function (secret) {
-      return cli.prompt('Endpoint Alias', {default: newEndpoint.hostname, timeout: 120000})
-      .then(function (alias) {
-        if (alias === 'default' || alias === 'cloud') {
-          throw new Error(`The word '${alias}' cannot be usded as an alias.`)
-        }
-        return saveEndpointOrError(newEndpoint, alias, secret)
-        .then(function (_) {
-          log(`Endpoint '${alias}' saved.`)
-        })
-      })
-    })
-    .catch(function (err) {
+    if (!secret)
+      secret = await cli.prompt('Endpoint Key', {type: 'hide', timeout: 120000})
+    if (!alias)
+      alias = await cli.prompt('Endpoint Alias', { default: newEndpoint.hostname, timeout: 120000 })
+
+    if (alias === 'default' || alias === 'cloud') {
+      throw new Error(`The word '${alias}' cannot be used as an alias.`)
+    }
+    return saveEndpointOrError(newEndpoint, alias, secret).then(function (_) {
```

Comment:
I don't know what is the pattern for async calls on fauna-shell but you can use the new `async/wait` syntax instead of promises if other places on codebase are using it.